lib/view_component/base.rb

```diff
@@ -50,6 +50,17 @@ def config
 
     attr_accessor :__vc_original_view_context
 
+    # Compoents can have a cache key that is used to cache the rendered output.
+    #
+    # @return [String]
+    def cache_key
```

Comment:
Can we separate this into a separate module, similar to how we implemented slots?

Reply:
on it

Reply:
done